### PR TITLE
Mining voucher redeem selection now uses a radial menu

### PR DESCRIPTION
**Pull request by Arkatos1**
## About The Pull Request

Mining voucher redeem selection is now using a radial menu when choosing a voucher rewards instead of an input menu. In addition to looking better, the menu now provides an additional information about specific reward set when a corresponding info button is clicked. Credits to wiki contributors for a part of the texts describing the kits from https://tgstation13.org/wiki/Shaft_Miner. Also cleans up and documents relevant code parts.

**Example image:**

RadialVoucher

## Why It's Good For The Game

Better selection method, less reliance on wiki sources to get information about voucher rewards.

## Changelog

:cl: Arkatos and wiki contributors
qol: Mining voucher redeem selection now uses a radial menu and provides additional information about the voucher sets. Literally years in the making, enjoy!
/:cl:

### CODE REVIEW

**Review by Fikou — CHANGES_REQUESTED** [review]
lovely!

**Review by Fikou — APPROVED** [review]
i meant as an approval oops

**Comment by MrMelbert** [comment]
Really nice. Also makes it super easy to add new kits (modularly even).

File: code/modules/mining/machine_vending.dm [hunk 1]
```diff
@@ -159,47 +159,61 @@
...
+ *
+ * * Arguments:
+ * * voucher The mining voucher that is being used to redeem the mining equipment
+ * * redeemer The mob that is redeeming the mining equipment
+ */
+/obj/machinery/mineral/equipment_vendor/proc/redeem_voucher(obj/item/mining_voucher/voucher, mob/redeemer)
+	var/static/list/set_types = list()
+	var/list/items = list()
+	for(var/datum/voucher_set/current_set as anything in subtypesof(/datum/voucher_set))
+		set_types[initial(current_set.name)] = new current_set
```
Comment (MrMelbert) [inline code/modules/mining/machine_vending.dm:181]: `set_types` is a static list but you seem to populate it with a full list of new `voucher_set` datums every time this proc is called.

It seems like `set_types` should only be setup once and the `items` list is the only thing that needs to be populated every redemption?

File: code/modules/mining/voucher_sets.dm [hunk 2]
```diff
@@ -0,0 +1,72 @@
+/datum/voucher_set
```
Comment (MrMelbert) [inline code/modules/mining/voucher_sets.dm:1]: Also if you could auto-doc this actual datum that would be nice

Comment (Arkatos1) [reply code/modules/mining/voucher_sets.dm:7]: Autodoc added.

File: code/modules/mining/machine_vending.dm (hunk 1 above)
Comment (Arkatos1) [reply code/modules/mining/machine_vending.dm]: You are right, I changed it so that the static list is populated only once now.

**Pull request merged**